filterUsers.js

```diff
@@ -0,0 +1,124 @@
+let users = [
+    {
+        userName: "Mario Rossi",
+        competenceUser: [
+            {
+                nameComp: 'java',
+                area: 'linguaggi programmazione',
+                level: 3
+            },
+            {
+                nameComp: 'Linux',
+                area: 'Sistemi operativi',
+                level: 2
+            }
+        ]
+    },
+    {
+        userName: "Franco Verdi",
+        competenceUser: [
+            {
+                nameComp: 'js',
+                area: 'linguaggi programmazione',
+                level: 4
+            },
+            {
+                nameComp: 'Windows',
+                area: 'Sistemi operativi',
+                level: 1
+            }
+        ]
+    }
+];
+
+
+
+
+
+function nameORCompetence() {
+    console.log(document.getElementById("initialChoice").value)
+    if (document.getElementById("initialChoice").value == "name") {
+        document.getElementById("searchByName").removeAttribute("hidden", "");
+    } else if (document.getElementById("initialChoice").value == "competenze") {
+        document.getElementById("searchByCompetence").removeAttribute("hidden", "");
+
+    } else if (document.getElementById("initialChoice").value == "competenzeLivello") {
+        document.getElementById("searchByCompetenceAndLevel").removeAttribute("hidden", "");
+
+    }
+    document.getElementById("DIVinitialChoice").setAttribute("hidden", "")
+}
+
+
+let printSpace = document.getElementById("show");
+
+function searchByName() {
+    clear(printSpace);
+    for (let index = 0; index < users.length; index++) {
+        if (users[index].userName.toLowerCase().includes(document.getElementById("nomeCercato").value.toLowerCase())) {
```

Comment:
estrai da questa condizione una variabile in cui inserisci il risultato booleano